language-server/modules/langserver-core/src/test/resources/completion/type_def/config/config7.json

```diff
@@ -24,14 +25,6 @@
       "insertText": "StrandData",
       "insertTextFormat": "Snippet"
     },
-    {
```

Comment:
This case tests the following completion scenario

```ballerina
type Cloneable readonly|xml|Cloneable[]|<cursor>

```
The LS uses the `visibleSymbols()` API in the semantic API to get the symbols list visible at a given cursor position. The `Cloneable` symbol  seems to be missing in the `visibleSymbols()` response with the current changes. Ballerina compiler allows a type definition to refer it self. Is this change intended? 

![image](https://github.com/ballerina-platform/ballerina-lang/assets/35211477/7f8b568e-fa83-40c7-bf91-c697b5854bb1)

@prakanth97 @hasithaa

Reply:
After this change symbol for the `Cloneable` will be created and added to the scope once all of its members resolved because, now we are following the depth first approach for type resolving.

Reply:
I think this is a valid use case. Since this is a type definition, we should see the type `Clonable` in RHS. The only challenge, in this case, is we are working on top of a partially completed source. Had a chat with @prakanth97, and we will see how to fix this.

Reply:
Updated 2310102e673453e99245aa4914cbf2bbe6de6e5a

Reply:
This is resolved isn't it ? @malinthar